### PR TITLE
[1LP][WIPTEST] Added test test_service_ansible_playbook_tagging

### PR DESCRIPTION
**Pull request by Yadnyawalkya**
{{pytest: cfme/tests/ansible/test_embedded_ansible_services.py::test_service_ansible_playbook_taging --long-running --use-provider complete}}

### CODE REVIEW

File: cfme/tests/ansible/test_embedded_ansible_services.py [hunk 1]
```diff
@@ -130,6 +131,32 @@ def test_service_ansible_playbook_crud(ansible_repository):
...
+            "create_new": True,
+            "provisioning_dialog_name": fauxfactory.gen_alphanumeric()
+        }
+    )
+    cat_item.create()
+    view = navigate_to(cat_item, 'Details')
+    view.policy.item_select("Edit Tags")
+    tag = ('Department', 'Support')
+    mixins.add_tag(tag)
+    assert view.smart_management.get_text_of("My Company Tags") == mixins.get_tags()[0]
```
Comment (quarckster) [inline cfme/tests/ansible/test_embedded_ansible_services.py]: **REQUIRED**
Please use the fixtures for the repository and the catalog item, do not duplicate the code. Also tags should be assigned via some method of `AnsiblePlaybookCatalogItem` instance.

File: cfme/services/catalogs/ansible_catalog_item.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 from utils.update import Updateable
 from utils.wait import wait_for
 from . import ServicesCatalogView
+from cfme.web_ui import mixins
```
Comment (quarckster) [inline cfme/services/catalogs/ansible_catalog_item.py]: **REQURIED**
Old web ui elements shouldn't be used, please use `WidgetasticTaggable` mixin.

File: cfme/tests/ansible/test_embedded_ansible_services.py [hunk 3]
```diff
@@ -130,6 +131,13 @@ def test_service_ansible_playbook_crud(ansible_repository):
     assert not cat_item.exists
 
 
+def test_service_ansible_playbook_tagvis(ansible_catalog_item):
+    tag = ('Department', 'Support')
+    ansible_catalog_item.add_tag(tag)
+    assert tuple(mixins.get_tags(tag="My Company Tags")[0].split(": ", 1)) == tag
+    ansible_catalog_item.remove_tag(tag)
```
Comment (quarckster) [inline cfme/tests/ansible/test_embedded_ansible_services.py]: **REQUIRED**
Please put it in `request.addfinalizer`

**Comment by dajoRH** [comment]
Lint report for commit 046c14ee0113481c353446de971fbc1da85d06ca:

`cfme/services/catalogs/ansible_catalog_item.py`:
- :heavy_exclamation_mark: Line 300:63: **W292** *no newline at end of file*

`cfme/tests/ansible/test_embedded_ansible_services.py`:
- :red_circle: Line 16:1: **F401** *'cfme.common.WidgetasticTaggable' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/common/__init__.py [hunk 4]
```diff
@@ -286,10 +286,12 @@ def get_tags(self, tenant="My Company Tags"):
             List of tags in format "Tag_category: Tag_name"
         """
         view = navigate_to(self, 'Details')
-        tags = view.tag.read()
+        tags = view.smart_management.read()
+        if isinstance(tags, dict):
+            tags = tags[tenant]
         if tags == 'No {} have been assigned'.format(tenant):
             return []
-        return filter(None, re.split(r'(.*?):\s*\S+\s?', tags))
```
Comment (Yadnyawalkya) [inline cfme/common/__init__.py]: It was capital ```S``` earlier, giving only ```[' ', 'Department', ' ']```,
After changing it to ```s```, giving ```[' ', 'Department', 'Support']``` which can be filter-out to obtain list !

File: cfme/common/__init__.py [hunk 5]
```diff
@@ -286,10 +286,12 @@ def get_tags(self, tenant="My Company Tags"):
             List of tags in format "Tag_category: Tag_name"
         """
         view = navigate_to(self, 'Details')
-        tags = view.tag.read()
+        tags = view.smart_management.read()
```
Comment (rlbabyuk) [inline cfme/common/__init__.py]: Please update class docs as 'tag' was changed to 'smart_management'. We need to get this clear, Details view should have 'smart_management', not 'tag'

File: cfme/tests/ansible/test_embedded_ansible_services.py [hunk 6]
```diff
@@ -130,6 +129,12 @@ def test_service_ansible_playbook_crud(ansible_repository):
     assert not cat_item.exists
 
 
+def test_service_ansible_playbook_tagvis(request, ansible_catalog_item):
```
Comment (rlbabyuk) [inline cfme/tests/ansible/test_embedded_ansible_services.py]: Fulfill tests steps, as for me it is not clear if tag visibility is tested or just tagging

File: cfme/common/__init__.py (hunk 5 above)
Comment (Yadnyawalkya) [inline cfme/common/__init__.py]: Here ```smart_management = SummaryTable("Smart Management")```   of ```DetailsAnsibleCatalogItemView``` can be used to retrieve smart_mangt table info. By replacing ```view.tag.read()``` to ```view.smart_management.read()``` can reduce duplication of code which possibly would happen for new ```tag = SummaryTable("Smart Management")``` widget.

Comment (Yadnyawalkya) [reply cfme/common/__init__.py]: sure, changing..

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d980acf7db5e56c99ff6765c9c6c1e46ab0cde4f:

`cfme/services/catalogs/ansible_catalog_item.py`:
- :bangbang: Line 204:101: **E501** *line too long (103 > 100 characters)*

Please, rectify these issues :smirk: .

* d980acf7db5e56c99ff6765c9c6c1e46ab0cde4f
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

File: cfme/services/catalogs/ansible_catalog_item.py [hunk 7]
```diff
@@ -191,15 +192,20 @@ class retirement(Tab):  # noqa
         info = SummaryForm("Retirement Info")
         variables_and_default_values = Table(".//div[@id='retirement']//table")
 
+
+class DetailsAnsibleCatalogItemView(ServicesCatalogView):
+    """Has to be in view standards, changed for WidgetasticTaggable.get_tags()"""
+    entities = View.nested(DetailsEntitiesAnsibleCatalogItemView)
```
Comment (Yadnyawalkya) [inline cfme/services/catalogs/ansible_catalog_item.py]: This ```DetailsEntitiesAnsibleCatalogItemView``` and ```entities```   has been added due to the fact that we have to use [PR5296#L296](https://github.com/mshriver/integration_tests/blob/e966eb6aa9bbac1d26c97e84f4219662411f99c5/cfme/common/__init__.py#L296) for get_tags() from new WidgetasticTaggable mixins

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit da72d993ff054bf5860188b3906b8f8f394a7531:

`cfme/services/catalogs/ansible_catalog_item.py`:
- :bangbang: Line 204:101: **E501** *line too long (103 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4a031f512c3a7db23000301d6de74dac1b9c5ef7:

`cfme/services/catalogs/ansible_catalog_item.py`:
- :bangbang: Line 204:101: **E501** *line too long (103 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 18000155d65365708ca00adab11a715a39a0d3b3:

`cfme/services/catalogs/ansible_catalog_item.py`:
- :bangbang: Line 204:101: **E501** *line too long (103 > 100 characters)*

Please, rectify these issues :smirk: .

* 18000155d65365708ca00adab11a715a39a0d3b3
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a11bcc2f7207b3bedc1dc12d9e8dbecdb1c10674:

`cfme/common/__init__.py`:
- :heavy_exclamation_mark: Line 756:33: **W292** *no newline at end of file*

`cfme/services/catalogs/ansible_catalog_item.py`:
- :bangbang: Line 204:101: **E501** *line too long (103 > 100 characters)*
- :heavy_exclamation_mark: Line 406:63: **W292** *no newline at end of file*

`cfme/tests/ansible/test_embedded_ansible_services.py`:
- :heavy_exclamation_mark: Line 270:51: **W292** *no newline at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit da13b28df905539e425898ba0f3832bdacda2e19:

`cfme/services/catalogs/ansible_catalog_item.py`:
- :bangbang: Line 204:101: **E501** *line too long (103 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/services/catalogs/ansible_catalog_item.py [hunk 8]
```diff
@@ -389,3 +395,12 @@ class Edit(CFMENavigateStep):
 
     def step(self):
         self.prerequisite_view.configuration.item_select("Edit this Item")
+
+
+@navigator.register(AnsiblePlaybookCatalogItem, 'EditTags')
+class EditTags(CFMENavigateStep):
+    VIEW = TagPageView
```
Comment (mshriver) [inline cfme/services/catalogs/ansible_catalog_item.py]: :+1:

File: cfme/tests/ansible/test_embedded_ansible_services.py [hunk 9]
```diff
@@ -147,6 +147,20 @@ def test_service_ansible_playbook_crud(ansible_repository):
...
+
+    Steps:
+        1. Login as a admin
+        2. Add tag for ansible_playbook
+        3. Check added tag
+        4. Remove the given tag
+    """
+    ansible_catalog_item.add_tag('Department', 'Support')
+    assert ansible_catalog_item.get_tags() == ['Department', 'Support']
+    request.addfinalizer(lambda: ansible_catalog_item.remove_tag('Department', 'Support'))
```
Comment (mshriver) [inline cfme/tests/ansible/test_embedded_ansible_services.py]: I don't think this should be done as a request finalizer, I think it should be part of the test. call remove_tag, and then assert get_tags() returns empty, or at least doesn't have `['Department', 'Support']`

**Comment by mshriver** [comment]
Expecting the test to uncollect in 5.7 because ansible config isn't valid there.

**Pull request merged**